python/lsst/daf/butler/core/config.py

```diff
@@ -540,16 +540,15 @@ def _splitIntoKeys(key):
             hierarchy = key.split(d)
             if temp:
                 hierarchy = [h.replace(temp, d) for h in hierarchy]
-            return hierarchy
+            # Copy the list to keep mypy quiet.
+            return list(hierarchy)
         elif isinstance(key, Iterable):
-            return list(key)
+            return [k for k in key]
```

Comment:
I was unsure about this one as well. It may be that it can go back to being `list(key)` (which I assume is more efficient?)

Reply:
`list` is likely more efficient, and it should work OK, I'll put it back.